php/php.dbgp/src/org/netbeans/modules/php/dbgp/breakpoints/ExceptionBreakpoint.java

```diff
@@ -36,6 +45,37 @@ public String getException() {
         return exceptionName;
     }
 
+    public synchronized void setExceptionMessage(String message) {
+        this.message = message;
+    }
+
+    @CheckForNull
+    public synchronized String getExceptionMessage() {
+        return buildText(message);
+    }
+
+    public synchronized void setExceptionCode(String code) {
+        this.code = code;
```

Comment:
OK, apparently we have more fields in this class :) But still, I believe that for `code` the same applies - it can be `volatile` and we should be fine. So, please verify the parent class, thanks.

Just to explain - these 2 fields (`message` and `code`) seem to be related, right? However, we set each of them in its own method, so there is no way (in this class) how to set both these fields atomically. If we wanted to do that, we would need something like:

```java
public synchronized void setMessageAndCode(String message, String code) {
    this.message = message;
    this.code = code;
}
```

In this case, we would need to use `synchronized` here and also whenever we would be reading any of these fields.

Reply:
Thanks for the explanation! Changed to using `volatile`.

Reply:
Nothing to thank for, really :)